src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1360,58 +1344,75 @@ CHIP_ERROR CASESession::OnFailureStatusReport(Protocols::SecureChannel::GeneralS
     return err;
 }
 
-CHIP_ERROR CASESession::IsResumptionRequestPresent(const System::PacketBufferHandle & msg, bool & resumptionRequested,
-                                                   ByteSpan & resumptionID, ByteSpan & resume1MIC)
+CHIP_ERROR CASESession::ParseSigma1(TLV::ContiguousBufferTLVReader & tlvReader, ByteSpan & initiatorRandom,
+                                    uint16_t & initiatorSessionId, ByteSpan & destinationId, ByteSpan & initiatorEphPubKey,
+                                    bool & resumptionRequested, ByteSpan & resumptionId, ByteSpan & initiatorResumeMIC)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-
-    System::PacketBufferTLVReader tlvReader;
-    TLV::TLVType containerType = TLV::kTLVType_Structure;
-
-    constexpr uint32_t kResumptionIDTag = 6;
-    constexpr uint32_t kResume1MICTag   = 7;
-
-    uint32_t lastDecodedTLVTag = 0;
-    bool resumptionIDTagFound  = false;
-    bool resume1MICTagFound    = false;
-
-    System::PacketBufferHandle msg_clone = msg.CloneData();
+    using namespace TLV;
+
+    constexpr uint8_t kInitiatorRandomTag    = 1;
```

Comment:
Might be a good idea also introduce a new method `CASESession::EncodeSigma1()` and use the same set of consts there.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/10091 on this.  Will do.